File: frontend/src/app/shared/models/conversation.ts
import { User } from './user';
import { ItemModel } from './item.model';
import { Message } from './message';

export interface Conversation {
  id: number;
  userId: number;
  user?: User;
  itemId?: number;
  item?: ItemModel;
  startedAt: Date | string;
  status: string;
  messages: Message[];
}

File: backend/Snagged.backend/Snagged.Domain/Entities/Conversation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Snagged.Domain.Entities
{
    public class Conversation
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User User { get; set; }
        public int? ItemId { get; set; }
        public Item Item { get; set; }
        public DateTime StartedAt { get; set; } = DateTime.Now;
        public string Status { get; set; } = string.Empty;
        
        
        public ICollection<Message> Messages { get; set; } = new List<Message>();
    }
}
